src/Resources/app/storefront/src/checkout/confirm-order.plugin.js

```diff
@@ -361,6 +361,9 @@ export default class ConfirmOrderPlugin extends Plugin {
     }
 
     renderPrePaymentButton(componentConfig, selectedPaymentMethodObject) {
+        if (selectedPaymentMethodObject.type === 'amazonpay') {
+            componentConfig.extra = this.setAddressDetails(componentConfig.extra);
```

Comment:
So will this config break the component if it isn't amazonpay? Thinking we could add it to all PrePayButton methods for simplicity if possible. I think it is only used for amazon at the moment anyway

Reply:
I'm not sure if it will break, but the `setAddressDetails` contains amazonpay specific logic, to change the `productType` to `PayOnly` if there is no phone number.